core-build/gulp-core-build-typescript/src/miniTasks/TslintCmdTask.ts

```diff
@@ -0,0 +1,155 @@
+import {
+  JsonFile,
+  IPackageJson
+} from '@microsoft/node-core-library';
+import * as fsx from 'fs-extra';
+import * as TSLint from 'tslint';
+
+import {
+  BaseCmdTask,
+  ICallCmdOptions
+} from './BaseCmdTask';
+
+/**
+ * @public
+ */
+export interface ITslintCmdTaskConfig {
+  /**
+   * The path to the tslint package if the task should override the version of tslint.
+   */
+  tslintPackagePath?: string;
+
+  /**
+   * Optional list of custom args to pass to "tslint"
+   */
+  customArgs?: string[];
+
+  /**
+   * The directory in which tslint should be invoked.
+   */
+  buildDirectory?: string;
+
+  /**
+   * If true, displays warnings as errors. Defaults to false.
+   */
+  displayAsError?: boolean;
+}
+
+/**
+ * @alpha
+ */
+export class TslintCmdTask extends BaseCmdTask<ITslintCmdTaskConfig> {
+  constructor() {
+    super(
+      'tslint',
+      {
+        displayAsError: false
+      }
+    );
+  }
+
+  public loadSchema(): Object {
+    return JsonFile.load(path.resolve(__dirname, '..', 'schemas', 'tslint-cmd.schema.json'));
+  }
+
+  public executeTask(gulp: typeof Gulp, completeCallback: (error?: string) => void): Promise<void> | undefined {
+    // require.resolve('tslint') resolves to <package>/lib/index.ts, and we want the package path
+    let tslintPackagePath: string = path.resolve(require.resolve('tslint'), '..', '..');
+    // Find the "tslint" executable
+    if (this.taskConfig.tslintPackagePath) {
+      // The tslint version is being overridden
+      if (!fsx.existsSync(this.taskConfig.tslintPackagePath)) {
+        completeCallback(
+          `The specified tslint tool path (${this.taskConfig.tslintPackagePath}) does not ` +
+          'exist'
+        );
+        return;
+      }
+
+      tslintPackagePath = this.taskConfig.tslintPackagePath;
+    }
+
+    const buildDirectory: string = this.taskConfig.buildDirectory || this.buildConfig.rootPath;
+
+    // Print the version
+    const packageJson: IPackageJson = JsonFile.load(path.join(tslintPackagePath, 'package.json'));
+    this.log(`TSLint version: ${packageJson.version}`);
+
+    const tslintBinaryPath: string = path.resolve(tslintPackagePath, 'bin', 'tslint');
+    if (!fsx.existsSync(tslintBinaryPath)) {
+      completeCallback('The tslint binary is missing. This indicates that tslint is not installed correctly.');
+      return;
+    }
+
+    const callCmdOptions: Partial<ICallCmdOptions> = {};
+    if (!this.taskConfig.displayAsError) {
+      callCmdOptions.onClose = (code: number, hasErrors: boolean, resolve: () => void) => {
+        resolve();
+      };
+    }
+
+    const customArgs: string[] = this.taskConfig.customArgs || [];
+    const additionalArgs: string[] = [];
+    if (
+      customArgs.indexOf('--formatters-dir') === -1 &&
+      customArgs.indexOf('-s') === -1 && // Shorthand for "--formatters-dir"
+      customArgs.indexOf('--format') === -1 &&
+      customArgs.indexOf('-t') === -1 // Shorthand for "--format"
```

Comment:
Add a note to customArgs docs that the warnings/errors will be busted if they change the formatter

Reply:
Should we even allow them to change the formatter.

---
In reply to: [199006726](https://github.com/Microsoft/web-build-tools/pull/729#discussion_r199006726) [](ancestors = 199006726)

Reply:
Yeah they should be able to change the formatter I think, but I added a note and a verbose log message.

---
In reply to: [199006877](https://github.com/Microsoft/web-build-tools/pull/729#discussion_r199006877) [](ancestors = 199006877,199006726)